libobs/util/utf8.c

```diff
@@ -146,8 +146,12 @@ size_t utf8_to_wchar(const char *in, size_t insize, wchar_t *out,
 	wlim = out == NULL ? NULL : out + outsize;
 
 	for (; p < lim; p += n) {
-		if (!*p)
+		if (!*p && !n)
```

Comment:
We should actually run through the rest of this function so it can correctly measure the length of the string. This function already handles nulls correctly so you can likely just remove the null check. If some other errors come up after removing the null check those should probably be fixed.

Reply:
I actually tried doing that before but it won't work. The previous function code handled the NULL case by breaking the loop thinking the string has ended once it sees one. So the rest of the code in loop does not handle the NULL case (logically). But we want to continue reading till we hit the file size by skipping over the NULLs and not including it in the file size. 
correct me if I am missing something here.

Reply:
Which part of the function does not handle nulls correctly? If its simple to fix maybe we can fix that as well. Otherwise skipping sounds fine.

Reply:
I think when we encounter a null, it moves inside the else block at line 180 and since `flag=0` it breaks and returns 0.

Reply:
Sorry it seems i misunderstood this fix, and we are instead just not copying out the null values in this function. Null (0x00) is treated like the other 7bit ascii characters in line 163 so it shouldnt error out. But it seems we were just skipping copying out the nulls with this fix, im less sure if this is what we would really want to do with this function but its probably workable as adding length handling everywhere ends up requiring many more changes.

Reply:
Oh yes line 163 handles the NULL case, I wonder where it breaks then cause the text does not get display correctly.